code/modules/projectiles/gun_helpers.dm

```diff
@@ -232,34 +232,37 @@ should be alright.
 //tactical reloads
 /obj/item/weapon/gun/MouseDrop_T(atom/dropping, mob/living/carbon/human/user)
 	if(istype(dropping, /obj/item/ammo_magazine))
-		var/obj/item/ammo_magazine/AM = dropping
-		if(!istype(user) || user.incapacitated(TRUE))
+		tactical_reload(dropping,user)
+	return..()
+
+///This performs a tactical reload with src using new_magazine to load the gun.
+/obj/item/weapon/gun/proc/tactical_reload(obj/item/ammo_magazine/new_magazine, mob/living/carbon/human/user)
+	var/obj/item/ammo_magazine/M = new_magazine
```

Comment:
M isnt much better, as it's used for mob
I reccomend just writing it out in future, but in this case you can just use new_magazine since that's already typecast